### PR TITLE
Fix flaky

### PR DESCRIPTION
**Pull request by humza1400**
# What is the purpose of this PR?
The main goal of this PR is to address and fix a flaky test found in the MiddlewaresTest.java. The test asShouldSerializeObjectAsJson intermittently fails due to inconsistencies in JSON parsing and validation.

# Why the test fails
The test fails intermittently because it attempts to parse the JSON string and extract integer and string values directly. This approach can lead to failures depending on the ordering and formatting of the JSON content. The process of extracting values from the JSON string wasn't robust enough to handle all scenarios and edge cases.

# How to reproduce the test failure
1. Clone the repository and checkout to the commit a03105ee1f98f8c5878f4e3c4d03c8732c6384bd.
2. Run the test suite multiple times or use a tool like NonDex to introduce randomness and observe the intermittent failures.
> mvn -pl apollo-api-impl edu.illinois:nondex-maven-plugin:2.1.1:nondex -Dtest=com.spotify.apollo.route.MiddlewaresTest#asShouldSerializeObjectAsJson

# Expected results
The test is expected to consistently pass, regardless of the ordering or formatting of the JSON string. The values extracted from the JSON should match the expected integer and string values.

# Actual results
The test intermittently fails, returning an assertion error. This inconsistency is due to the non-deterministic nature of the JSON string's ordering and content. This leads to scenarios where the extracted values are not as expected

# Description of fix
We introduced a more reliable method of parsing and validating the JSON string. We leveraged Jackson, a JSON processing library already included in the project dependencies, to deserialize the JSON string into a Java object. This approach ensures that the extraction of values is done in a more type-safe and reliable manner, preventing failures related to JSON string inconsistencies.

### CODE REVIEW

File: apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java [hunk 1]
```diff
@@ -133,19 +135,29 @@ public void asShouldNotSerializeResponseByteString() throws Exception {
...
+ @Test
   public void asShouldSerializeObjectAsJson() throws Exception {
-    class TestData {
-      public String theString = "hi";
-      public int theInteger = 42;
-    }
+  class TestData {
+  public String theString = "hi";
+  public int theInteger = 42;
+  }
```
Comment (winglam) [inline apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java:143]: The changes from Line 138 to Line 143 seem unnecessary. Can you undo them?

File: apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java [hunk 2]
```diff
@@ -133,19 +135,29 @@ public void asShouldNotSerializeResponseByteString() throws Exception {
...
 
-    serializationFuture.complete(new TestData());
+  TestData testData = new TestData();
 
-    //noinspection ConstantConditions
-    String json =
-        getResult(Middlewares.autoSerialize(serializationDelegate)).payload().get().utf8();
-    assertThat(json, equalToIgnoringWhiteSpace("{\"theString\":\"hi\",\"theInteger\":42}"));
+  serializationFuture.complete(testData);
```
Comment (winglam) [inline apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java:147]: The changes above also seem unnecessary.

File: apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java [hunk 3]
```diff
@@ -133,19 +135,29 @@ public void asShouldNotSerializeResponseByteString() throws Exception {
...
+
+  //noinspection ConstantConditions
+  String json =
+  getResult(Middlewares.autoSerialize(serializationDelegate)).payload().get().utf8();
+
+  ObjectMapper mapper = new ObjectMapper(); // Use Jackson's ObjectMapper to parse JSON
+  JsonNode jsonNode = mapper.readTree(json); // Parse the JSON string into a JsonNode
```
Comment (winglam) [inline apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java:154]: You can collapse Line 150 to 154 into one line as you do not use the intermediary variables more than once.

File: apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java [hunk 4]
```diff
@@ -43,6 +43,8 @@
 
 import okio.ByteString;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (winglam) [inline apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java:47]: Can you please put these imports on top of Line 44?

File: apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java [hunk 5]
```diff
@@ -133,19 +135,29 @@ public void asShouldNotSerializeResponseByteString() throws Exception {
...
+
+  //noinspection ConstantConditions
+  String json =
+  getResult(Middlewares.autoSerialize(serializationDelegate)).payload().get().utf8();
+
+  ObjectMapper mapper = new ObjectMapper(); // Use Jackson's ObjectMapper to parse JSON
+  JsonNode jsonNode = mapper.readTree(json); // Parse the JSON string into a JsonNode
+
+  String actualString = jsonNode.get("theString").asText();
+  int actualInteger = jsonNode.get("theInteger").asInt();
```
Comment (winglam) [inline apollo-api-impl/src/test/java/com/spotify/apollo/route/MiddlewaresTest.java:157]: You shouldn't need to create these variables. Please just have these lines inside the assert statement.

**Review by winglam — COMMENTED** [review]
Great job on creating this PR. 

I highly encourage you to address the comments I pointed out, email me again once you have done so, and if everything looks good then, we can have you open the PR to the developers so you can actually help improve open source code!
